components/common/src/util/path.rs

```diff
@@ -92,58 +87,47 @@ async fn interpreter_paths() -> Result<Vec<PathBuf>> {
-                                             INTERPRETER_COMMAND, &path);
-                                    return Err(Error::FileNotFound(path));
-                                }
-                            }
-                        }
-                        None => {
-                            install::type_erased_start(&mut ui::NullUi::new(),
-                                                       &default_bldr_url(),
-                                                       &ChannelIdent::stable(),
-                                                       &(ident.clone(),
-                                                         PackageTarget::active_target())
-                                                                                        .into(),
-                                                       &*PROGRAM_NAME,
-                                                       VERSION,
-                                                       FS_ROOT_PATH.as_path(),
-                                                       &cache_artifact_path(None::<String>),
-                                                       None,
-                                                       &InstallMode::default(),
-                                                       &LocalPackageUsage::default(),
-                                                       InstallHookMode::default()).await?;
-                            let pkg_install =
-                                PackageInstall::load(&ident, Some(FS_ROOT_PATH.as_ref()))?;
-                            pkg_install.paths()?
-                        }
-                    }
+                    if install::type_erased_start(&mut ui::NullUi::new(),
+                                                &default_bldr_url(),
+                                                &ChannelIdent::stable(),
+                                                &(ident.clone(),
+                                                    PackageTarget::active_target())
+                                                                                .into(),
+                                                &*PROGRAM_NAME,
+                                                VERSION,
+                                                FS_ROOT_PATH.as_path(),
+                                                &cache_artifact_path(None::<String>),
+                                                None,
+                                                &InstallMode::default(),
+                                                &LocalPackageUsage::default(),
+                                                InstallHookMode::default()).await.is_err() {
+                                                    return Err(Error::InterpreterNotFound(ident));
+                                                }
+                    let pkg_install = PackageInstall::load(&ident, Some(FS_ROOT_PATH.as_ref()))?;
```

Comment:
Can we use the `PackageInstall` returned from `install::typed_erased_start` or is it important to try and `load` the package?

Reply:
No that is a great point and I just did not notice that the previous code was not taking advantage of that